internal/services/contracts_events_consumer.go

```diff
@@ -145,3 +153,51 @@ func (c *ContractsEventsConsumer) setPrivilegeHandler(e *ContractEventData) erro
 
 	return nil
 }
+
+type MintedAftermarketDeviceArgs struct {
+	TokenID       string `mapstructure:"newTokenId"`
+	DeviceAddress string
+	Owner         string // TODO - confirm who the minter is
+}
+
+func (c *ContractsEventsConsumer) setMintedAfterMarketDevice(e *ContractEventData) error {
+	p := MintedAftermarketDeviceArgs{}
+	err := mapstructure.WeakDecode(e.Arguments, &p)
+	if err != nil {
+		return err
+	}
+
+	autopiApiService := NewAutoPiAPIService(c.settings, c.db.DBS)
+	device, err := autopiApiService.GetDeviceByEthAddress(p.DeviceAddress)
+	if err != nil {
+		c.log.Error().Msg(fmt.Sprintf("Couldn't fetch dongle with eth_address %s.", p.DeviceAddress))
+		return fmt.Errorf("couldn't fetch dongle with eth_address %s", p.DeviceAddress)
+	}
+
+	var maybeAddr null.Bytes
+
+	if strAddr := device.EthereumAddress; common.IsHexAddress(p.DeviceAddress) {
+		maybeAddr = null.BytesFrom(common.FromHex(strAddr))
+	} else {
+		c.log.Warn().Str("address", device.EthereumAddress).Msg("Invalid device Ethereum address from AutoPi.")
+	}
+
+	tokenID, _ := new(big.Int).SetString(p.TokenID, 16)
```

Comment:
Is it in base 16? That would be somewhat surprising to me. I can show you how to play with some of the real output from contract-event-processor. This is probably the most serious comment here.

If we're going to check that the address is valid we might as well make sure that this works.

We can probably bypass `big.Int` and go straight to `decimal.Big`.